.github/workflows/ci.yml

```diff
@@ -17,16 +17,89 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
           cache: 'npm'
-      - run: npm ci
+      - uses: actions/cache@v3
+        id: npm-cache
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm
+      - name: Install dependencies if cache invalid
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: npm ci
+  pr:
+    needs: install
+    strategy:
+      matrix:
+        node: [16.x, 18.x]
+        os: [macos-latest, ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      VITE_FR_AM_URL: ${{ secrets.VITE_FR_AM_URL }}
+      VITE_FR_AM_COOKIE_NAME: ${{ secrets.VITE_FR_AM_COOKIE_NAME }}
+      VITE_FR_OAUTH_PUBLIC_CLIENT: ${{ secrets.VITE_FR_OAUTH_PUBLIC_CLIENT }}
+      VITE_FR_REALM_PATH: ${{ secrets.VITE_FR_REALM_PATH }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Restore npm dependencies
+        uses: actions/cache@v3
+        id: npm-cache
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm
       - run: npm run build
-        env:
-          VITE_FR_AM_URL: ${{ secrets.VITE_FR_AM_URL }}
-          VITE_FR_AM_COOKIE_NAME: ${{ secrets.VITE_FR_AM_COOKIE_NAME }}
-          VITE_FR_OAUTH_PUBLIC_CLIENT: ${{ secrets.VITE_FR_OAUTH_PUBLIC_CLIENT }}
-          VITE_FR_REALM_PATH: ${{ secrets.VITE_FR_REALM_PATH }}
-      - name: Publish to Chromatic
-        if: ${{ matrix.os == 'ubuntu-latest' && matrix.node < 17 && !github.event.pull_request.draft }}
-        uses: chromaui/action@v1
-        with:
-          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+      - name: Build Storybook
+        run: npm run build:storybook --quiet
+      - run: npx playwright install
+      - name: Serve Storybook and run tests
+        if: ${{ runner.os == 'ubuntu-latest' }} # this one hangs on mac os for some reason
+        run: |
+          npx concurrently -k -s first -n "SB,TEST" -c "magenta,blue" \
+            "npx http-server storybook-static --port 6006 --silent" \
+            "npx wait-on tcp:6006 && npm run test:storybook"
+  visual:
```

Comment:
Can you explain what the `visual` job is for please?

Reply:
It's just running the visual tests that storybook has - the ones that above, hang on mac.

Reply:
Understood, thanks for explaining.

Reply:
I mis-spoke, and edited the above comment to reflect hte correct wording. It is the chromatic test run. 

We can absolutely add the label requirement, i'd probably move it out of this workflow and into its own, if we prefer only running the tests when it has a label.

Reply:
I'd like to ask for a name change since "visual" is misleading. I consider these "integration tests" or "interaction tests". We could even just name it "chromatic". Just not a fan of "visual".